### PR TITLE
Tug can pull a wheeled cart

### PR DESCRIPTION
**Pull request by EnterpriseProgrammer**
## About the PR 
Adds two OR statements to allow tug vehicles to pull a single wheeled cart. Flatbeds are not able to be attached to the back of a wheeled cart.

I imagine that tugs could be used to move the emergency equipment already set up (Medbay, Engineering repairs, Forensics) more quickly, either if mechanics makes one for the departments or if more tugs are included in the maps. Either way, this change allows for more tug utility beyond delivering cargo crates.

## Why's this needed? 
Adds to the utility of the tug and further completes the logic for wheeled emergency carts of different purposes.

## Changelog

```changelog
(u)Xyzzy:
(+)Tugs can now pull wheeled carts.
```

### CODE REVIEW

**Comment by ZeWaka** [comment]
Please follow the proper changelog format, otherwise it won't get picked up.

File: code/WorkInProgress/tug.dm [hunk 1]
```diff
@@ -20,21 +20,22 @@
...
-				return
-			else if (src.next_cart == connecting)
-				src.next_cart = null
-				user.visible_message("[user] disconnects [connecting] from [src].", "You disconnect [connecting] from [src].")
-				return
-			else
-				user.show_text("\The [src] already has a cart connected to it!", "red")
-				return
+		if (istype(C, /obj/tug_cart) || istype(C, /obj/storage/cart) && in_range(C, src))
+			if (istype(C, /obj/tug_cart) || istype(C, /obj/storage/cart))
```
Comment (ZeWaka) [inline code/WorkInProgress/tug.dm:24]: umm what, why is there two of these????

**Review by EnterpriseProgrammer — COMMENTED** [review]
Removed the redundant double-if at Line 23/24.

**Comment by EnterpriseProgrammer** [comment]
Welp, apparantly that double if statement is necessary for some reason, according to the checks.

File: code/WorkInProgress/tug.dm [hunk 2]
```diff
@@ -20,21 +20,24 @@
...
```
Comment (ZeWaka) [inline code/WorkInProgress/tug.dm:24]: No, it's not necessary, you're just not grasping that DM is indentation sensitive and therefore you can't just remove if statements without adjusting the inner body.

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/WorkInProgress/tug.dm [hunk 3]
```diff
@@ -20,7 +20,7 @@
 			user.show_text("\The [C] is too heavy for \the [src]!", "red")
 			return
 
-		if (istype(C, /obj/tug_cart) && in_range(C, src))
+		if (istype(C, /obj/tug_cart) || istype(C, /obj/storage/cart) && in_range(C, src))
 			var/obj/tug_cart/connecting = C
```
Comment (ZeWaka) [inline code/WorkInProgress/tug.dm:24]: this will runtime on line 27 due to a miscast

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**